### PR TITLE
add Kronecker sequences, with R2 and K21-2 irrationals

### PR DESCRIPTION
**Pull request by drone115b**
Kronecker sequences with R2 and K21-2 irrationals, and documentation for them.

build-doc.sh failed for me because of a missing template.

### CODE REVIEW

**Comment by dcoeurjo** [comment]
Very interesting PR!, thanks @drone115b

**Comment by dcoeurjo** [comment]
(For the doc, you'd need the material theme: `pip3 install mkdocs-material`)

File: src/samplers/SamplerKronecker/Kronecker.cpp [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+const double Kronecker::R4[4] = {
+  0.85667488385450287,
+  0.73389185662712599,
+  0.62870672103780863,
+  0.53859725722361005
+};
+
+const double Kronecker::K21_2[2] = {
+  std::sqrt((long double)506598872547596.0)/(long double)29147227.0,
```
Comment (dcoeurjo) [inline src/samplers/SamplerKronecker/Kronecker.cpp:52]: My compiler is complaining about the `long double` to `double` cast in intializer lists. Ok for you if I do explicit cast ?: 

```suggestion
  static_cast<double>(std::sqrt((long double)506598872547596.0)/(long double)29147227.0),
```

File: src/samplers/SamplerKronecker/Kronecker.cpp [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+  std::sqrt((long double)107882942223468.0)/(long double)28993644.0
```
Comment (dcoeurjo) [inline src/samplers/SamplerKronecker/Kronecker.cpp:53]: the same here after

File: src/samplers/SamplerKronecker/Kronecker.cpp (hunk 1 above)
Comment (drone115b) [reply src/samplers/SamplerKronecker/Kronecker.cpp]: I updated my branch to do this static cast, thanks.

File: src/samplers/SamplerKronecker/Kronecker.cpp (hunk 2 above)
Comment (drone115b) [reply src/samplers/SamplerKronecker/Kronecker.cpp]: I updated my branch to do this static cast, thanks.

**Comment by drone115b** [comment]
Thanks @dcoeurjo and happy new year.  I updated my branch to do the static cast that you suggested.
I tested a build of the docs and the new pages in this branch look correct.  I have not included any of the changes under docs/ as part of this PR - I assume there is some automation done upon merge.

**Comment by dcoeurjo** [comment]
Perfect, thanks a lot
(and no ;) at this point there is no automation or GitHub action to deploy the doc.. I should do something like that).

**Comment by dcoeurjo** [comment]
and happy new year  @drone115b

**Comment by dcoeurjo** [comment]
@drone115b thanks again for the PR. What is the current status of the paper "Optimizing Kronecker Sequences for Multidimensional Sampling" et JCGT?

**Pull request merged**